mobile/library/swift/cxx/EnvoyHeaders.swift

```diff
@@ -0,0 +1,42 @@
+@_implementationOnly import EnvoyCxxSwiftInterop
+
+typealias EnvoyHeaders = [String: [String]]
+
+func toSwiftHeaders(_ headers: envoy_headers) -> EnvoyHeaders {
+  defer {
+    release_envoy_headers(headers)
+  }
+
+  return (0..<Int(headers.length)).reduce(into: EnvoyHeaders()) { result, i in
+    let header = headers.entries[i]
+    if
+      let headerKey = String.fromEnvoyData(header.key),
+      let headerValue = String.fromEnvoyData(header.value)
+    {
+      result[headerKey, default: []].append(headerValue)
+    }
+  }
+}
+
+func toEnvoyHeaders(_ headers: EnvoyHeaders?) -> envoy_headers {
```

Comment:
I think that our current naming scheme is to call these to toNativeHeaders (or swift_toNativeHeaders if there is name collision). What do you think?